### ISSUE TITLE
Team dropped my item before disconnecting for 5mins

### ISSUE THREAD

**Issue by shinkirouzero** [ValveSoftware/Dota2-Gameplay#7059]
### Item name

Ring of Health, Hood of Defiance, Magic Wand

### Description

I disconnected from the game and reconnected right away. But my teammate dropped my item.
Got disconnected at 11:05. Showed a 5mins left to Reconnect message. Got Reconnected right away.
Got disconnection again at 13:26. No message on how many minutes left to reconnect. 
My teammate dropped my item at 13:36.
I reconnected at 13:43 and realized that my items are gone.
 I am not disconnected for more than 5 mins. How did this happen?

https://user-images.githubusercontent.com/126167777/220914912-e5c66ea9-b4bc-4624-b9c6-0a2d150fcac1.mp4

Match ID: 7029999484

### Example Match ID and Timestamp

7029999484 - 13:26

**Comment by commentator24** [comment]
i watched the replay a bit, perhaps after using wand teammates able to drop items? I dont know what caused it. but i noticed both you and earth spirit mouse movements are the kind i would report in overwatch guilty for scripting. perhaps better you dont use the script anymore before you 2 are caught. maybe the script interrupt with inventory?

**Comment by shinkirouzero** [comment]
> i watched the replay a bit, perhaps after using wand teammates able to drop items? I dont know what caused it. but i noticed both you and earth spirit mouse movements are the kind i would report in overwatch guilty for scripting. perhaps better you dont use the script anymore before you 2 are caught. maybe the script interrupt with inventory?



Yes, after that wand, Earth spirit is able to drop my item. 
I am not using any scripts. There is a 10% to 30% packet loss during the game, maybe that is why our cursor is teleporting.

**Comment by commentator24** [comment]
well.. I think valve hasn't yet caught cheating behaviours like this one, but I played over 24,000 matches so I can easily tell apart abnormal mouse movements like these mouse shakes and acceleration that snap on last hit targets; odd mouse positions and camera snap to low hp teammates and I reported these as scripting in overwatch. perhaps I'm wrong but I don't know any settings that do these. if there are settings that do these mouse behaviours i'd like to learn too..

https://user-images.githubusercontent.com/109324155/220997737-a83787dc-9478-44a4-8c0b-40bc48617c07.mp4
try follow exact mouse movements in the video.. would you find it hard to follow / do you find the mouse positions are very odd?

for example in the video you're constantly right clicking to move.  at :033 you can see mouse shakes, 0:37 odd mouse position right clicking to move way past the unit to last hit and at 0:32 and 0:44 mouse accelerates and snap to target to try deny / last hit. and this is just first minute example.

**Comment by shinkirouzero** [comment]
> 

Here are my settings,

Windows 10 with cursor speed set to 15

I'm using the auto-repeat right mouse button.

Using Smart Attack to attack the unit nearest to the cursor and not the unit closest to my hero
Disable auto attack when Stop is held.
I'm using auto-attack and spamming the "S" stop command. I will stop pressing the S button when I'm about to deny or hit.
Sometimes I hold the Stop key instead of spamming it, and after releasing it, my hero will perform an attack.
I do move the cursor when I'm deciding to last hit or to deny a creep, making it look like the cursor is shaking.
The creeps are just close to each other. With a faster curser speed, small hand movements are enough to change target.
I could not believe such quick and precise mouse movements could be suspected of using scripts.
Feel free to watch my other replays. You may see that I'm struggling to deny or last hit when enemies are competing with me on the lane.

**Comment by sare167** [comment]
With everything in mind I'm looking for blog spots that talk about sonnets and found yours by accident. This post is very informative, and you've posted a very important and long one, which has helped me learn a lot. I know you'll keep doing it, and that the news we get from you will be very important and helpful. Definitely, thank you so much for writing such a great post. [dordle](https://dordle-game.com)

**Comment by commentator24** [comment]
do you mean the setting is like this?
![image](https://user-images.githubusercontent.com/109324155/221104941-70523664-56c5-42b1-8fe6-8586ff20f6a9.png)
auto attack is always and stop "S"? i've tried this demo hero but the cursor does not snap on targets nor do the mouse shakes like in 0:33-0:36, plus my cursor turns red when I have to select and right click a target, but in the video yours stay green and constantly right clicking but never red to try attack, which seems odd to me, how do you select target if you don't issue attack command?

also the odd mouse positions. did you try to follow exact mouse movements in the video? like at 
0:37 your cursor is not actually positioned to last hit that juicy creep after a deny, but way past over it to accelerate on ranged creep? 
![image](https://user-images.githubusercontent.com/109324155/221113462-89da3e3e-0ed0-4234-b228-c0904d9d94e4.png)
0:38 you click left, then up, then down, then bottom left as soon as luna enters map? i just don't get it. and at 0:43 you were ready to deny then accelerate to last hit. when I see these kind of abnormal mouse movements I immediately report as scripting as it looks pre-recorded / assisted movements to me. anyway thanks for trying to help me understand how you play.. sorry I just dont get it. 😓 i'm not here to get you or anything since there is no report button for scripting when watching replays and im too lazy to report on steam for suspected cheating (i only do it if it happened in my games or in overwatch), so if you're not banned thats great. I was wrong to say you're cheating.. and if you're innocent but caught bad luck it's not me okay 😆 and until I learn how to do similar mouse behaviours I will continue to report odd mouse behaviours like this one as scripting in overwatch replays.

**Comment by shinkirouzero** [comment]
I understand that you have concerns about my gameplay and that you believe I am cheating by using scripts. However, I want to assure you that I am not cheating in any way. My mouse movements and keyboard inputs are the result of many hours of practice with the aid of the settings that I am using. Furthermore, the video you uploaded does not prove that I am using scripts to get gold by performing finishing hits on enemy creatures. In fact, I can assure you that I struggle at times to get the last hit and obtain the gold.

Here are videos to compare the actual recording of the play and the video in the replay. Take your time to compare them. You may see that I am sending a quick attack on the actual recording. However, on the replay, the quick attack animation is not displayed. It only shows my cursor moving in different directions.

https://user-images.githubusercontent.com/126167777/221164932-ac36d425-fc65-40aa-a151-8c1f16f55327.mp4


https://user-images.githubusercontent.com/126167777/221164965-9256da9d-6f89-4074-bb94-6cada1474811.mp4

In another video, with the use of "Smart Attack," I may target creeps for an attack even if my cursor is way past the creep. The cursor is also displayed in red, and there is a red circle on the target creep. 
 Compare it to the replay. The replay barely shows that I'm sending a smart attack. All you can see are green move commands. I am able to issue a command to perform a last hit when my cursor is way past the enemy creep with the aid of Smart Attack and perform it as soon as I hit the attack button because of the quick attack. As


https://user-images.githubusercontent.com/126167777/221165006-e42216ee-a476-4128-add9-310fb6d0c45f.mp4


https://user-images.githubusercontent.com/126167777/221165074-28e968b2-8079-4a05-b03f-afae2a820b3b.mp4

Just like what you have asked: "I have to select and right click a target, but in the video yours stay green and constantly right clicking but never red to try attack, which seems odd to me, how do you select target if you don't issue attack command?"

With your very long experience in Dota 2, I am pretty sure that you are knowledgeable about the fact that the replay and actual recording of the play are different. Packet loss greatly affects the information that will be displayed on the replay. With packet loss, you will notice that your hero might not execute some of the mouse and keyboard commands. With the definition of packet loss itself, "... one or more packets of data travelling across a computer network fail to reach their destination." (Wikipedia).
Some of the mouse and keyboard commands do not reach the server and are not recorded in the replay. I already told you that there is a 10% to 30% packet loss during that game. It is also the main reason I disconnected more than once in the game.

It seems that you are not knowledgeable about the fact that the replay and actual game recording will display differently. Actual recordings of the play show attack and move commands differently.

If I am actually cheating, it may be very unwise for me to expose myself as a cheater. I have nothing to gain from cheating, and I take my integrity as a player very seriously.

I can no longer wonder why so many innocent players were banned because of misinterpretations. I doubt your experience as a participant in Overwatch in the Dota 2 community. I already exposed my public matchmaking data to everyone. Feel free to watch my other replays.

I stopped playing for a long time for work. I created this new account because my younger brother tremendously reduced the MMR and behavior score of my main account. As an average Dota 2 player in the low rank tier, I feel sad to be accused of cheating. Here is the user ID of my main account: (407981743). Watch my games and take time to analyze them. You are also free to submit a ticket and report me as a cheater. I suggest that you get a third-party opinion, such as from a moderator or another player.
This person can review my gameplay and determine if I am cheating or not.

My concern here is to report the issue about my teammates dropping my items early in the game. I hope you can understand my position and accept my assurance that I am not cheating. Have a good day sir.

**Comment by commentator24** [comment]
no i don't care if you cheat or not, unless you're in my games, which I would definitely report suspected cheating. I also don't care if im wrong as I'm here to learn and only comment what I observed. and if innocent players accused as cheating it's up to valve developers to take action or not. this is a bug issue forum not for catching players scripting / ban appeals anyway so I doubt developer personally take action.

Im merely suggesting perhaps if you do use script (from what I observed with your odd mouse behaviours, mouse snaps on targets and mouse shakes) perhaps that interfered with inventory. on that note, i observed many mouse snaps and accelerations in 7029999484 but not in 7031364024 where mouse behaviours also very different.

no i didnt know replay and actual game recording will display differently because I sometimes still see attack commands like this one, but now i see it only happen when picking up runes in replays. thanks for clearing that up for me with your videos.
![image](https://user-images.githubusercontent.com/109324155/221374075-1c141b91-a784-4577-a979-acf3fa3542b0.png)

and initially i thought this is also attack command as shown by the PA: mostly red and sometimes green
![image](https://user-images.githubusercontent.com/109324155/221374123-86a5c237-1eb6-4a3a-b1e1-4ce8fae672e4.png)

whereas I don't get why yours is the other way around: mostly green and sometimes red. initially I thought it may have something to do with scripting where actual mouse location and assisted movements are different, but perhaps its just wrong shader? issue (not sure what it's called). and sometimes i thought i saw attack command like in 1:20 (red circle on your spectre) and still have no idea why.
![image](https://user-images.githubusercontent.com/109324155/221374330-eabb2fac-0698-4383-b979-75b943a1d6f2.png)

perhaps if you have an idea what caused these you could help me learn more and report it as selection shader? issue. not sure what it's called lol.
